opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -266,6 +270,21 @@ namespace Opm {
         return treeUpdated;
     }
 
+    void Schedule::handleWHISTCTL(const ParseContext& parseContext, const DeckKeyword& keyword) {
+        for( const auto& record : keyword ) {
+            const std::string& cmodeString = record.getItem("CMODE").getTrimmedString(0);
+            WellProducer::ControlModeEnum controlMode = WellProducer::ControlModeFromString( cmodeString );
+            m_controlModeWHISTCTL = controlMode;
+            const std::string& bhp_terminate = record.getItem("BPH_TERMINATE").getTrimmedString(0);
```

Comment:
(just saw this accidentally, I am not reviewing this)

It is actually ok and not a bug to use a const reference in that context, the returned object will then have the same lifetime as the reference. You may already know this of course, sorry for interrupting!